src/Microsoft.AspNet.OData.Shared/TypeHelper.cs

```diff
@@ -416,11 +416,9 @@ internal static Type GetImplementedIEnumerableType(Type type)
         // This code is copied from DefaultHttpControllerTypeResolver.GetControllerTypes.
         internal static IEnumerable<Type> GetLoadedTypes(IWebApiAssembliesResolver assembliesResolver)
         {
-            List<Type> result = new List<Type>();
-
             if (assembliesResolver == null)
             {
-                return result;
+                yield return null;
```

Comment:
>yield return null; [](start = 16, length = 18)

Is this the same as returning an empty list, or does it return a list with a null entry?

Reply:
yield return null is returning a list with one item (null) in the list.

```C#
        static void Main(string[] args)
        {
            var types = GetLoadedTypes();
            Console.WriteLine(types.Count());
            Console.WriteLine(types.ElementAt(0) == null ? "null" : "not null");
        }

        internal static IEnumerable<Type> GetLoadedTypes()
        {
            yield return null;
        }
```

Returns:
```C#
1
null
``

---
In reply to: [598852813](https://github.com/OData/WebApi/pull/2436#discussion_r598852813) [](ancestors = 598852813)

Reply:
That was my assumption.  But that's not the semantics we want here, right? Don't we want to return an empty list for this case?

---
In reply to: [598898800](https://github.com/OData/WebApi/pull/2436#discussion_r598898800) [](ancestors = 598898800,598852813)

Reply:
I changed the codes as discussed.